include/pipettes/core/tasks/i2c_task.hpp

```diff
@@ -33,18 +33,52 @@ class I2CMessageHandler {
     void visit(ReadFromI2C &m) {
         i2c_device.central_receive(m.buffer.data(), m.buffer.size(), m.address,
                                    TIMEOUT);
-        m.client_callback(m.buffer);
+        m.handle_buffer(m.buffer);
+        m.client_callback();
     }
 
-    void visit(PollReadFromI2C &m) {
+    void visit(SingleRegisterPollReadFromI2C &m) {
+        // TODO (lc, 03-01-2022): see about making this non-blocking
+        // TODO (lc, 03-01-2022): we should try to consolidate polling to
+        // support any number of registers potentially.
         auto empty_array = m.buffer;
         for (int i = 0; i < m.polling; i++) {
+            i2c_device.central_transmit(m.buffer.data(), m.buffer.size(),
+                                        m.address, TIMEOUT);
             i2c_device.central_receive(m.buffer.data(), m.buffer.size(),
                                        m.address, TIMEOUT);
-            m.client_callback(m.buffer);
+            m.handle_buffer(m.buffer);
             m.buffer = empty_array;
             i2c_device.wait_during_poll(m.delay_ms);
```

Comment:
we also might want to delay this to another pr but this is not how you want to do fixed-frequency polling since it adds the time taken for the message to the wait. If you pass 100ms here, then you'll be reading every 100ms + transmit/receive time. Usually what you want to do is take a timestamp before transmission begins and then use something like [`vTaskDelayUntil`](https://www.freertos.org/vtaskdelayuntil.html) to delay until `timestamp+100ms`, which gives you more consistent timing (or use a timer)

Reply:
Put in a TODO comment for a follow-up PR.